### PR TITLE
Fixes issue with DTL Environment where build libs may not have updated

### PR DESCRIPTION
**Pull request by nigurr**
Issue: DTL Register Environment has new flag which is required to be passed by DTA Task.
           However build agent might have got latest DTA Task with the old libraries, which can break the build with the below error. 
          ##[error]A parameter cannot be found that matches parameter name 'Persist'.
          ##[error]Cannot bind argument to parameter 'MachineEnvironment' because it is null.
Fix:   Check if the member exists in the DLL, if it exists call the new flow otherwise call the old flow.

### CODE REVIEW

**Comment by nigurr** [comment]
@muthuk, please assign someone from your team, ask them to verify the fix and take into production.

File: Tasks/DeployVisualStudioTestAgent/DeployTestAgent.ps1 [hunk 1]
```diff
@@ -23,6 +23,11 @@ Function CmdletHasMember($memberName) {
     return $cmdletParameter
 }
 
+Function RegisterEnvCmdletHasMember($memberName){
+    $cmdletParameter = (gcm Register-Environment).Parameters.Keys.Contains($memberName)
```
Comment (KrishnaAdityaB) [inline Tasks/DeployVisualStudioTestAgent/DeployTestAgent.ps1]: any reason why you didn't use ContainsKey() instead of Keys.Contains()?

**Comment by KrishnaAdityaB** [comment]
looks good

**Comment by pryenugu** [comment]
Looks good.

**Pull request merged**